orca_detector/live_feed_listener.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: future_fstrings -*-
+
+"""
+Script to listen to a live audio feed. The audio can either be played in the system audio output, or saved to a file.
+
+W251 (Summer 2019) - Spyros Garyfallos, Ram Iyer, Mike Winton
+"""
+
+import click
+import glob
+import m3u8
+import numpy as np
+import orca_params
+import os
+import random
+import shutil
+import time
+from threading import Thread
+import uuid
+import urllib.request
+from database_parser import extract_segment_features
+from inference import create_network
+
+
+def _save_audio_segments(stream_url,
+                         stream_name,
+                         segment_seconds,
+                         iteration_seconds,
+                         mix_with,
+                         output_path,
+                         verbose):
+    """
+    Uses ffmpeg (via CLI) to retrieve audio segments from audio_url
+    and saves it to the local output_path.
+
+    If mix_with exists, it mixes this file with the live feed source.
+    The result 
+    """
+
+    file_name = f"{stream_name}_%02d.wav"
+    output_file = os.path.join(output_path, file_name)
+    mix_with = ''
+
+    if os.path.exists(mix_with):
+        mix_with = '-i {} -filter_complex amix=inputs=2:duration=first'.format(
+            (mix_with))
+
+    ffmpeg_cli = 'ffmpeg -y -i {} {} -t {} -f segment -segment_time {} {}'.format(
+        (stream_url), (mix_with), (iteration_seconds), (segment_seconds), (output_file))
+
+    if not verbose:
+        ffmpeg_cli = ffmpeg_cli + ' -loglevel warning'
+    os.system(ffmpeg_cli)
+
+
+def _perform_inference(model, encoder, inference_samples_path):
```

Comment:
Yes, but by default it's only live feed noise only. It will contain mixed noise with a positive sample if you drop a sample file in the positive samples folder.